src/defrag.c

```diff
@@ -297,8 +297,8 @@ Defrag4Reassemble(ThreadVars *tv, DefragTracker *tracker, Packet *p)
     rp->recursion_level = p->recursion_level;
 
     int fragmentable_offset = 0;
-    int fragmentable_len = 0;
-    int hlen = 0;
+    uint16_t fragmentable_len = 0;
+    uint16_t hlen = 0;
     int ip_hdr_offset = 0;
```

Comment:
Wonder why this needs to stay `int` here, but can be made `uint16_t` in IPv6. In general I'm not clear why it needs to be signed and why it needs to be bigger than `uint16_t`. Does it take the offset from the whole packet? Not just the IP part?

@jasonish any idea?

Reply:
I think a previous version of the PR did this, and showed issues in QAlab runs.

Reply:
I do not think previous versions of the PR did this. I can try in a next one.
I just fixed the warnings, this `ip_hdr_offset` is not cast back to another uint16, just used as a pointer offset and in an addition for `SET_PKT_LEN`

For IPv6, it is also used in `unfragmentable_len` computation which is an uint16 (and does not exist in IP4)